woocommerce-gateway-stripe.php

```diff
@@ -217,6 +217,11 @@ public function init() {
 					add_filter( 'woocommerce_get_sections_checkout', [ $this, 'filter_gateway_order_admin' ] );
 				}
 
+				// If UPE is enabled, initialize the Gateway here for the hooks in the constructor to work.
+				if ( WC_Stripe_Feature_Flags::is_upe_checkout_enabled() ) {
+					new WC_Stripe_UPE_Payment_Gateway();
```

Comment:
Creating the gateway object here will mean it is created twice in most page loads and it will register and run each action twice. Once here when the stripe plugin is initialized and once whenever the WC_Payment_Gateways class is constructed.

I think we should really keep the ajax actions in a separate class. We could keep the UPE actions separate if we want but I don't think it's necessary. If these ajax actions just need access to the gateway object they should just use `WC_Stripe_Intent_Controller::get_gateway()` which will retrieve a gateway object and construct it if necessary but since it uses the core WC_Payment_Gateways class it will only create one gateway object. 

Also moving these methods increases the size of the gateway class.

Reply:
For context, when we started with the UPE migration the idea was to keep the implementation as similar to WC Payments as possible to help maintainability... moving the hooks was briefly discussed in this PR https://github.com/woocommerce/woocommerce-gateway-stripe/pull/1659#discussion_r693129179.

I don't mind keeping the hooks in separate files, I just find confusing that `WC_Stripe_Intent_Controller ` has `save_upe_appearance_ajax` and `clear_upe_appearance_transient` which are not related to intents, and `WC_Stripe_Order_Handler` has `add_action( 'wp', [ $this, 'maybe_process_redirect_order' ] );` which is not order specific. I would like @FangedParakeet's input here, as he was the one who raised this issue.

I can change the scope of the PR to just replace the `$gateway = new WC_Stripe_UPE_Payment_Gateway();` with:
```
$gateway = $this->get_gateway();
if ( ! is_a( $gateway, 'WC_Stripe_UPE_Payment_Gateway' ) ) {
    return;
}
```
and copy the `get_gateway()` to `WC_Stripe_Order_Handler`. Which will solve the problem of UPE javascripts being included even when UPE is not enabled.

Reply:
I agree with @superdav42 here. I suggested moving the hooks in the first place because I was concerned that there may be some issues with creating multiple instances of `WC_Stripe_UPE_Payment_Gateway` inside the `WC_Stripe_Intent_Controller` class. It looks like `WC_Stripe_Intent_Controller::get_gateway` has been refactored since I've last noticed it and now solves this problem by fetching an instance of the gateway class, using the core `WC_Payment_Gateways` class, instead of creating a new instance of `WC_Stripe_UPE_Payment_Gateway`. It makes sense to me to continue to take advantage of this new and improved `get_gateway` function, by just moving all of the AJAX actions back here.

> I don't mind keeping the hooks in separate files, I just find confusing that WC_Stripe_Intent_Controller has save_upe_appearance_ajax and clear_upe_appearance_transient which are not related to intents, and WC_Stripe_Order_Handler has add_action( 'wp', [ $this, 'maybe_process_redirect_order' ] ); which is not order specific. 

This is definitely also a valid point. Maybe, since this has now become more of an Action centre than an Intent hub, `WC_Stripe_Intent_Controller` might be in line for a rename--something like `WC_Stripe_Action_Controller`, I'd hazard to suggest, perhaps? Also, could we move that action on the `wp` hook and `maybe_process_redirect_order` into `WC_Stripe_Intent_Controller` (or whatever we end up calling it), using `get_gateway`, as well? That sounds like it would solve all these problems and be a little less organizationally confusing all at once.

Reply:
Cool, I'll change the PR to move all hooks to  `WC_Stripe_Intent_Controller`... and we can rename it later, maybe to`WC_Stripe_Controller` as there is no other controller 🤷🏼 ; but we can evaluate that later.